src/SFML/Graphics/Shader.cpp

```diff
@@ -235,19 +235,19 @@ struct Shader::UniformBinder
 
 
 ////////////////////////////////////////////////////////////
-Shader::Shader() : m_shaderProgram(0), m_currentTexture(-1), m_textures(), m_uniforms()
+void Shader::ShaderProgramDeleter::operator()(const unsigned int shaderProgram)
 {
+    TransientContextLock lock;
+
+    // Destroy effect program
+    if (shaderProgram)
+        glCheck(GLEXT_glDeleteObject(castToGlHandle(shaderProgram)));
 }
 
 
 ////////////////////////////////////////////////////////////
-Shader::~Shader()
+Shader::Shader() : m_shaderProgram(0, {}), m_currentTexture(-1), m_textures(), m_uniforms()
```

Comment:
This is taking ownership of a shader program identified with 0.
Default initialization would make more sense, meaning the shader does not own a shader program yet.